runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2140,15 +2185,28 @@ bool TR_EscapeAnalysis::checkDefsAndUses(TR::Node *node, Candidate *candidate)
                   TR::Node *useNode = _useDefInfo->getNode(useIndex+_useDefInfo->getFirstUseIndex());
                   int32_t useNodeVN = _valueNumberInfo->getValueNumber(useNode);
                   int32_t i;
+                  bool valueNumberSeen = false;
+
                   for (i = candidate->_valueNumbers->size()-1; i >= 0; i--)
                      {
                      if (candidate->_valueNumbers->element(i) == useNodeVN)
-                        break;
+                        {
+                        valueNumberSeen = true;
+
+                        // Only add this value number if it's not to be ignored
+                        if (!_ignoreableUses->get(candidate->_valueNumbers->element(i)))
```

Comment:
Henry caught this _ignoreableUses is currently working in terms of node global indices so this check needs to be fixed before this is merged.

Reply:
Please let me know when all changes you want to merge with this PR are done and I will run fresh testing on it.

Reply:
This is now fixed in commit https://github.com/eclipse/openj9/pull/5737/commits/bfef953a7de444ebe4a9204198142a316d393f26